src/OpenTelemetry/Internal/OpenTelemetrySdkEventSource.cs

```diff
@@ -197,12 +200,18 @@ public void NoDroppedExportProcessorItems(string exportProcessorName, string exp
             this.WriteEvent(31, exportProcessorName, exporterName);
         }
 
+#if NET6_0_OR_GREATER
```

Comment:
IMO, it is easier to just add these attributes as `internal` with a shared source file. That way the other TFMs can see them.

We do that for ASP.NET (https://github.com/dotnet/aspnetcore/blob/main/src/Shared/TrimmingAttributes.cs) and I did the same for IdentityModel (https://github.com/AzureAD/azure-activedirectory-identitymodel-extensions-for-dotnet/blob/dev/src/Common/TrimmingAttributes.cs).

That way you don't need all these ugly `#if`s when referencing these attributes.

Reply:
You can conditionally compile this file into the project when it doesn't referencing net6. Like this:

```xml
  <ItemGroup>
    <Compile Include="..\Common\TrimmingAttributes.cs" LinkBase="Common"
             Condition="'$(TargetFrameworkIdentifier)' != '.NETCoreApp'" />
  </ItemGroup>
```

Reply:
addressed your comments!

Reply:
Updated as suggested